developer/developer-guide.md

```diff
@@ -1,3 +1,872 @@
+        }
+    }
+  ]
+}
+```
+
+* Nesting
+
+Operators can be nested to form complex queries:
+```json
+{
+  "and" : [
+    {
+      "not": [
+         {
+            "property" : {
+                "name" : "example:customProperty",
+                "value" : "#FFF233"
+             }
+          }
+      ]
+    },
+    {
+      "or": [
+          {
+            "property" : {
+                "name" : "example:customProperty",
+                "value" : "#FFF233"
+              }
+          },
+          {
+            "property" : {
+              "name" : "example:customProperty",
+              "value" : "#AA3BF5"
+            }
+          }
+      ]
+    }
+  ]
+}
+```
+##### Search terms
+
+Search terms can be split into two groups: special terms and generic terms.
+Special terms are used to search for specific concepts, generic terms are used to search for property values at large.
+
+###### Special terms
+
+* type
+
+`Supports transactional metadata queries`
+
+Lookup for any nodes of the given contentmodel type.
+```json
+{ "type": "cm:folder" }
+```
+
+
+* aspect
+
+`Supports transactional metadata queries`
+
+Lookup for any nodes which have the give aspect.
+```json
+{ "aspect":  "sys:folder"}
+```
+
+
+* noderef
+
+`Supports transactional metadata queries`
+
+Lookup for a specific alfresco noderef.
+```json
+{ "noderef": "workspace://SpacesStore/c4ebd508-b9e3-4c48-9e93-cdd774af8bbc" }
+```
+
+
+* path
+
+`Supports transactional metadata queries`
+
+Lookup for nodes at the given XPath.
+```json
+{ "path": "/app:company_home/cm:Fred_x0020_Performance_x0020_Test/*" }
+```
+
+
+* parent
+
+`Supports transactional metadata queries`
+
+Lookup for any nodes with the given node as parent. Takes a noderef as argument.
+```json
+{ "parent": "workspace://SpacesStore/c4ebd508-b9e3-4c48-9e93-cdd774af8bbc" }
+```
+
+* text
+
+Lookup for any nodes that have given text in their content. Requires content of nodes to be indexed by solr to be found.
+```json
+{ "text": "xenit solutions" }
+```
+
+* category
+
+Lookup for any nodes are of the given category.
+```json
+{ "category": "/cm:generalclassifiable/cm:Software_x0020_Document_x0020_Classification/member" }
+```
+
+* all
+
+Lookup for any nodes with a hit for the searchterm in any field or in the content.
+```json
+{ "all":  "Xenit solutions"}
+```
+
+* isunset
+
+Lookup for any nodes where the value of given propety is not set.
+```json
+{ "isunset":  "cm:author"}
+```
+
+* exists
+
+Lookup for any nodes where the given property is present.
+```json
+{ "exists": "cm:author" }
+```
+
+* isnull
+
+Lookup for any nodes where the value of the given property is set to null.
+```json
+{ "isnull":  "cm:author"}
+```
+
+* isnotnull
+
+Lookup for any nodes where the value of the given property is set and not null.
+```json
+{ "isnotnull": "cm:author" }
+```
+
+###### Note Unsupported terms
+
+The following terms are available in alfresco, but are currently not supported by Alfred-api:
+* isroot
```

Comment:
This list does not render. Please review the PDF yourself too before asking a code review. 

Probably fixed by extra leading newline.